lib/pbench/test/functional/server/test_datasets.py

```diff
@@ -91,6 +97,10 @@ def test_upload_all(self, server_client: PbenchServerClient, login_user):
                 t = tarballs[dataset.name]
                 assert dataset.name in dataset.metadata["server.tarball-path"]
                 assert dataset.metadata["dataset.operations"]["UPLOAD"]["state"] == "OK"
+                assert (
+                    dataset.metadata["dataset.metalog.pbench.script"]
+                    == dataset.metadata["server.benchmark"]
+                )
```

Comment:
While this assertion (generally) holds at the moment, this isn't really what we want going forward, right?  That is, we're hoping that at some point we'll have `pbench-user-benchmark` as the script and `fio` as the benchmark, right?  So, it would be better if we could structure this assertion in a way which doesn't break when we get where we want to be going.  (I.e., "test as specification".)

(Also, you didn't take this shortcut at lines 190 and 192.)

Reply:
For the archiveonly test, I know exactly what values to expect; this is an iteration, and all we can really know at this point is that they ought to match.

When and if we work out a more general "base benchmark" recognition that doesn't rely on `dataset.metalog.pbench.script`, we'll have to figure out how to mimic that on the functional test side.

Reply:
> all we can really know at this point is that they ought to match.

They should only match currently because of an implementation detail, not because that's specifically the way we want these fields to behave.  It would be best if the tests were constructed in a way that doesn't require them to change when details of the implementation change....  (Ideally, we would not change the tests nearly as often as we seem to do.)